Aml 1415 add apprentice task

robinoconnor opened the pull request.

Commit:
Added more processor to handle previously unsupported messages

Commit:
Added missing topic attributes and missing message processor for tasks

Commit:
Names changes added

arvinthseran commented:
Pending work in this PR
- [ refactor CohortCreated class from CohortCreatedMessageProcessor.cs] 
- [ Update commitments events Nuget package in Tasks]

arvinthseran committed:
update commitments event nuget package

arvinthseran committed:
refactored CohortCreated processor and updated tests

arvinthseran reviewed (approved):
👍 Lets go for it

Commit:
Added ability to complete all existing tasks with one call

- This is to support the story which requires a one time completion of outstanding tasks

Commit:
Added check for add apprentices task creation

- We only create the task if no cohorts have been added already

Commit:
Added new message handling code for cohort approval rejection

Commit:
Updated logs so they are more unique

Commit:
Adding missing subscription attribute

arvinthseran commented on src/SFA.DAS.Tasks.Worker/MessageProcessors/ApprenticeshipUpdateAcceptedMessageProcessor.cs:
```diff
@@ -26,7 +26,7 @@ public ApprenticeshipUpdateAcceptedMessageProcessor(IMessageSubscriberFactory su
         protected override async Task ProcessMessage(ApprenticeshipUpdateAccepted message)
         {
-            _logger.Debug($"Completing 'apprentice changes to review' task for account id {message.AccountId}, " +
+            _logger.Debug($"Aprretniceship update accepted. Completing 'apprentice changes to review' task for account id {message.AccountId}, " +
```
Apprenticeship?
